### ISSUE TITLE
Investigate slow termination checking

### ISSUE THREAD

**Issue by romac** [epfl-lara/stainless#126]
I am noticing random tests failing in the termination test suite. For example, the `NNFSimple` benchmark in [this build](http://laraquad4.epfl.ch:9000/epfl-lara/stainless/build/2059), which disappeared when I tried to reproduce it locally, but instead it is the `BinarySearch` test that failed.

**Comment by mantognini** [comment]
With #132 (commit 0feb5107134e298b24ee38ea8065f8e8e52a3ec9) we disabled the NNFSimple test for termination. This commit should be reverted once the cause of termination checking slowness is identified and fixed.

**Comment by samarion** [comment]
I believe this is a broader issue with parallelism during regressions. I don't know why it's happening, but the tests when run sequentially on my machine are individually much faster than when run on larabot (about a factor 3 I'd say). The full regression still takes much longer when run sequentially obviously but I don't see why the tests are so much slower, also when compared to when Stainless wasn't running in parallel.

**Comment by samarion** [comment]
Seems to be mostly fixed by https://github.com/epfl-lara/stainless/commit/74ac842db46899312d19d8abc5a31d3e1b26d95b
We maybe still need to fix the timeout on decreases checks but I'll leave that for another issue if it turns out to be relevant.
